src/app/ui/timeline.h

```diff
@@ -291,6 +300,12 @@ namespace app {
 
     AniControls m_aniControls;
 
+    bool m_celPreview;
+    gfx::Rect m_celPreviewOverlayRect;
+    // TODO sortable/cleanable by MRU
+    typedef std::map< doc::ObjectId, base::SharedPtr<she::Surface> > ThumbCache;
```

Comment:
`ThumbCache` might be a separate class instead a typedef in other file outside the timeline. There is already a `ThumbnailGenerator` to generate thumbnails for file names, so this `ThumbCache` should be something different.